### ISSUE TITLE
Regarding the update of augnet parameters

### ISSUE THREAD

**Issue by bregmangh** [BUserName/Learning_to_diversify#3]
Hi, thanks for answering my questions one after another. Now, I have two questions: 
1. By analyzing the code, I think the following code will also update the parameters of augnet network after execution：self.noise_lv，self.shift_var*，self.shift_mean*. This seems to be in contradiction with equation (12) in the paper?
loss = class_loss + self.args.alpha2 * likeli + self.args.alpha1 * con  
loss.backward()

2. The following code will also update the parameters of the augnet network after execution: self.noise_lv，self.shift_var*，self.shift_mean*. This seems to be in contradiction with equation (13) in the paper?
(dist + self.args.beta * div).backward()  
self.convertor_opt.step()

**Comment by bregmangh** [comment]
sorry, the second question should be:
2. The following code will also update the parameters of the "extractor" network after execution. This seems to be in contradiction with equation (13) in the paper?
(dist + self.args.beta * div).backward()
self.convertor_opt.step()
